### ISSUE TITLE
Swap deployment fails - Pod isn't starting or can't be found

### ISSUE THREAD

**Issue by antontroshin** [telepresenceio/telepresence#1477]
### What were you trying to do?

`telepresence --swap-deployment my-service`

### What did you expect to happen?

Command executed successfully

### What happened instead?

Connection to telepresence failed to be established.
Same cluster, and same namespace, command `telepresence --expose 8080 --expose 9990` works fine.
Not using VPN

### Automatically included information

Command line: `['/usr/local/bin/telepresence', '--swap-deployment', 'my-service']`
Version: `0.108-10-gd4bda22`
Python version: `3.9.0 (default, Oct 27 2020, 14:15:17) 
[Clang 12.0.0 (clang-1200.0.32.21)]`
kubectl version: `Client Version: v1.19.3 // Server Version: v1.18.9-eks-d1db3c`
oc version: `(error: [Errno 2] No such file or directory: 'oc')`
OS: `Darwin Antons-MBP-2 19.6.0 Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64 x86_64`

```
Traceback (most recent call last):
  File "/usr/local/bin/telepresence/telepresence/cli.py", line 135, in crash_reporting
    yield
  File "/usr/local/bin/telepresence/telepresence/main.py", line 65, in main
    remote_info = start_proxy(runner)
  File "/usr/local/bin/telepresence/telepresence/proxy/operation.py", line 271, in act
    wait_for_pod(runner, self.remote_info)
  File "/usr/local/bin/telepresence/telepresence/proxy/remote.py", line 140, in wait_for_pod
    raise RuntimeError(
RuntimeError: Pod isn't starting or can't be found: {'conditions': [{'lastProbeTime': None, 'lastTransitionTime': '2020-11-24T15:10:01Z', 'status': 'True', 'type': 'Initialized'}, {'lastProbeTime': None, 'lastTransitionTime': '2020-11-24T15:10:01Z', 'message': 'containers with unready status: [my-service]', 'reason': 'ContainersNotReady', 'status': 'False', 'type': 'Ready'}, {'lastProbeTime': None, 'lastTransitionTime': '2020-11-24T15:10:01Z', 'message': 'containers with unready status: [my-service]', 'reason': 'ContainersNotReady', 'status': 'False', 'type': 'ContainersReady'}, {'lastProbeTime': None, 'lastTransitionTime': '2020-11-24T15:10:01Z', 'status': 'True', 'type': 'PodScheduled'}], 'containerStatuses': [{'containerID': 'docker://8a48fc82a0fc87c177263a98f91da66fb147983e8900cd2546b54baf2942514d', 'image': 'datawire/telepresence-k8s:0.108', 'imageID': 'docker-pullable://datawire/telepresence-k8s@sha256:4090b197f6c4eafe5591b9fba065fb89478b0e2b675d5994fd879d5799cbb06c', 'lastState': {}, 'name': 'my-service', 'ready': False, 'restartCount': 0, 'started': False, 'state': {'running': {'startedAt': '2020-11-24T15:10:07Z'}}}], 'hostIP': '10.190.249.220', 'phase': 'Running', 'podIP': '10.190.250.120', 'podIPs': [{'ip': '10.190.250.120'}], 'qosClass': 'Burstable', 'startTime': '2020-11-24T15:10:01Z'}

```

Logs:

```
-dev --namespace test-namespace get pod my-service-5ea153286fd24bf6ae5975aa4522e593 -o json
 185.2 TEL | [75] captured in 1.64 secs.
 185.5 TEL | [76] Capturing: kubectl --context default --namespace test-namespace get pod my-service-5ea153286fd24bf6ae5975aa4522e593 -o json
 185.6 TEL | [77] Running: sudo -n echo -n
 185.7 TEL | [77] ran in 0.12 secs.
 187.2 TEL | [76] captured in 1.79 secs.
 187.5 TEL | [78] Capturing: kubectl --context default --namespace test-namespace get pod my-service-5ea153286fd24bf6ae5975aa4522e593 -o json
 189.2 TEL | [78] captured in 1.71 secs.
 189.5 TEL | [79] Capturing: kubectl --context default --namespace test-namespace get pod my-service-5ea153286fd24bf6ae5975aa4522e593 -o json
 191.1 TEL | [79] captured in 1.63 secs.
 191.3 TEL | [80] Capturing: kubectl --context default --namespace test-namespace get pod my-service-5ea153286fd24bf6ae5975aa4522e593 -o json
 192.9 TEL | [80] captured in 1.59 secs.
 192.9 TEL | END SPAN remote.py:109(wait_for_pod)  182.2s

```

Logs from telepresence pod
```
Retrieving this pod's namespace from the process environment
Pod's namespace is 'test-namespace'
Listening...
2020-11-24T15:10:07+0000 [-] Loading ./forwarder.py...
2020-11-24T15:10:07+0000 [-] /etc/resolv.conf changed, reparsing
2020-11-24T15:10:07+0000 [-] Resolver added ('172.20.0.10', 53) to server list
2020-11-24T15:10:07+0000 [-] SOCKSv5Factory starting on 9050
2020-11-24T15:10:07+0000 [socks.SOCKSv5Factory#info] Starting factory <socks.SOCKSv5Factory object at 0x7f9fb0ced3c8>
2020-11-24T15:10:07+0000 [-] DNSDatagramProtocol starting on 9053
2020-11-24T15:10:07+0000 [-] Starting protocol <twisted.names.dns.DNSDatagramProtocol object at 0x7f9fb0ced668>
2020-11-24T15:10:07+0000 [-] Loaded.
2020-11-24T15:10:07+0000 [twisted.scripts._twistd_unix.UnixAppLogger#info] twistd 20.3.0 (/usr/bin/python3.6 3.6.8) starting up.
2020-11-24T15:10:07+0000 [twisted.scripts._twistd_unix.UnixAppLogger#info] reactor class: twisted.internet.epollreactor.EPollReactor.
2020-11-24T15:10:38+0000 [Poll#error] Failed to contact Telepresence client:
2020-11-24T15:10:38+0000 [Poll#error] An error occurred while connecting: 99: Address not available.
2020-11-24T15:10:38+0000 [Poll#warn] Perhaps it's time to exit?
2020-11-24T15:11:08+0000 [Poll#error] Failed to contact Telepresence client:
2020-11-24T15:11:08+0000 [Poll#error] An error occurred while connecting: 99: Address not available.
2020-11-24T15:11:08+0000 [Poll#warn] Perhaps it's time to exit?
2020-11-24T15:11:38+0000 [Poll#error] Failed to contact Telepresence client:
2020-11-24T15:11:38+0000 [Poll#error] An error occurred while connecting: 99: Address not available.
2020-11-24T15:11:38+0000 [Poll#warn] Perhaps it's time to exit?
2020-11-24T15:12:08+0000 [Poll#error] Failed to contact Telepresence client:
2020-11-24T15:12:08+0000 [Poll#error] An error occurred while connecting: 99: Address not available.
2020-11-24T15:12:08+0000 [Poll#warn] Perhaps it's time to exit?
…
```

**Comment by deicon** [comment]
Seems to be related to https://github.com/telepresenceio/telepresence/issues/1467
Once I remove all Probes, the Pod can be swapped as usual.

**Comment by antontroshin** [comment]
@deicon thanks a lot, just tested and can confirm that removing `startupProbe` "fixes" the problem.
In addition, having both `livenessProbe` and `readinessProbe` doesn't seem to be problematic.
However, removing `startupProbe` from the `Deployment` manifest is not ideal and a proper solution would be gladly appreciated. Hope one can be found.

**Comment by deicon** [comment]
@antontroshin True. Removing the probes from the actual deployment is not an option at all. I created a PR here, to properly extend telepresence to exclude the startupProbe as well during swap deployment

**Comment by antontroshin** [comment]
Closing, issue fixed in #1479
